sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorCoordinator.java

```diff
@@ -0,0 +1,512 @@
+
+    private volatile Exception lastKnownLinkError;
+    private volatile Instant lastKnownErrorReportedAt;
+
+    ReactorCoordinator(Sender sender, SendLinkHandler handler, ReactorProvider reactorProvider,
+        MessageSerializer messageSerializer, Duration timeout, AmqpRetryPolicy retry) {
+
+        this.sender = sender;
+        this.handler = handler;
+        this.reactorProvider = reactorProvider;
+
+        this.messageSerializer = messageSerializer;
+        this.retry = retry;
+        this.timeout = timeout;
+        this.subscriptions = Disposables.composite(
+            this.handler.getDeliveredMessages().subscribe(this::processDeliveredMessage),
+
+            this.handler.getLinkCredits().subscribe(credit -> {
+                logger.verbose("Credits on link: {}", credit);
+                this.scheduleWorkOnDispatcher();
+            }),
+
+            this.handler.getEndpointStates().subscribe(
+                state -> {
+                    logger.verbose("[{}] Connection state: {}", sender.getName(), state);
+                    this.hasConnected.set(state == EndpointState.ACTIVE);
+                    endpointStateSink.next(AmqpEndpointStateUtil.getConnectionState(state));
+                }, error -> {
+                    logger.error("[{}] Error occurred in coordinator endpoint handler.", sender.getName(), error);
+                    endpointStateSink.error(error);
+                }, () -> {
+                    endpointStateSink.next(AmqpEndpointState.CLOSED);
+                    endpointStateSink.complete();
+                    hasConnected.set(false);
+                }),
+
+            this.handler.getErrors().subscribe(error -> {
+                logger.error("[{}] Error occurred in coordinator error handler.", getLinkName(), error);
+                endpointStateSink.error(error);
+            }),
+            () -> { }
+        );
+    }
+
+
+    /**
+     * Create AMQP {@link Message} for completing the transaction.
+     *
+     * @param transaction to be used for completing the transaction.
+     * @param isCommit for commiting or rollback
+     * @return created AMQP {@link Message}.
+     */
+    private Message getCompleteTransactionMessage(AmqpTransaction transaction, boolean isCommit) {
+        Message message = Proton.message();
+        Discharge discharge = new Discharge();
+        discharge.setFail(!isCommit);
+        discharge.setTxnId(new Binary(transaction.getTransactionId().array()));
+        message.setBody(new AmqpValue(discharge));
+        return message;
+    }
+
+    /**
+     * Create AMQP {@link Message} for creating the transaction.
+     *
+     * @return created AMQP {@link Message}.
+     */
+    private Message getCreateTransactionMessage() {
+        Message message = Proton.message();
+        Declare declare = new Declare();
+        message.setBody(new AmqpValue(declare));
+        return message;
+    }
+
+    //@Override
```

Comment:
Override?

Reply:
Why isCommit? I would expect completing the transaction to be commiting a transaction.

Reply:
A transaction needs to be completed with either commit or rollback. SB uses this flag to wither commit or rollback this transaction.